### PR TITLE
Upload Leishen lslidar_c16 source code into apollo

### PR DESCRIPTION
**Pull request by tongsky723**
upload lslidar_c16 apollo code

### CODE REVIEW

**Comment by ycool** [comment]
Thanks a lot for the contribution. Can you help to click CLAassistant  to sign CLA?

File: modules/drivers/lslidar_c16/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# toplevel CMakeLists.txt for a catkin workspace
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/CMakeLists.txt]: 这个文件是catkin_init_workspace时生成的软链，无需提交

File: modules/drivers/lslidar_c16/lslidar_c16_compensator/include/lslidar_c16_compensator/const_variables.h [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    TOPIC_PREFIX + "compensator/PointCloud2";
+
+/**
+* @brief Order array for re-ordering point cloud.
+* Refer to Velodyne official manual
+*/
+const int ORDER_16[16] = {0, 2, 4, 6, 8, 10, 12, 14, 1, 3, 5, 7, 9, 11, 13, 15};
+
+const int ORDER_64[64] = {38, 39, 42, 43, 32, 33, 36, 37, 40, 41, 46, 47, 50,
+                          51, 54, 55, 44, 45, 48, 49, 52, 53, 58, 59, 62, 63,
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/lslidar_c16_compensator/include/lslidar_c16_compensator/const_variables.h]: 这些参数是velodyne lidar专有的，应该删除

File: modules/drivers/lslidar_c16/lslidar_c16_decoder/include/lslidar_c16_decoder/lslidar_c16_decoder.h [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+static const int PACKET_SIZE        = 1206;
+static const int BLOCKS_PER_PACKET  = 12;
+static const int PACKET_STATUS_SIZE = 4;
+static const int SCANS_PER_PACKET =
+        (SCANS_PER_BLOCK * BLOCKS_PER_PACKET);
+static const int FIRINGS_PER_PACKET =
+        FIRINGS_PER_BLOCK * BLOCKS_PER_PACKET;
+
+// Pre-compute the sine and cosine for the altitude angles.
+static const double scan_altitude[16] = {
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/lslidar_c16_decoder/include/lslidar_c16_decoder/lslidar_c16_decoder.h]: 常量需要全大写

File: modules/drivers/lslidar_c16/lslidar_c16_decoder/include/lslidar_c16_decoder/lslidar_c16_decoder.h [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    std::sin(scan_altitude[10]), std::sin(scan_altitude[11]),
+    std::sin(scan_altitude[12]), std::sin(scan_altitude[13]),
+    std::sin(scan_altitude[14]), std::sin(scan_altitude[15]),
+};
+
+typedef struct{
+    double distance;
+    double intensity;
+}point_struct;
+
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/lslidar_c16_decoder/include/lslidar_c16_decoder/lslidar_c16_decoder.h]: 使用c++方式进行定义，如下方PointXYZIT

File: modules/drivers/lslidar_c16/lslidar_c16_decoder/src/lslidar_c16_decoder.cpp [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+    pnh.param<double>("frequency", frequency, 20.0);
+    pnh.param<bool>("publish_point_cloud", publish_point_cloud, true);
+    pnh.param<bool>("publish_channels", publish_channels, true);
+    pnh.param<bool>("apollo_interface", apollo_interface, false);
+    pnh.param<string>("fixed_frame_id", fixed_frame_id, "map");
+    pnh.param<string>("child_frame_id", child_frame_id, "lslidar");
+
+    angle_base = M_PI*2 / point_num;
+
+    if (apollo_interface)
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/lslidar_c16_decoder/src/lslidar_c16_decoder.cpp]: if 最好加上{}

File: modules/drivers/lslidar_c16/lslidar_c16_decoder/src/lslidar_c16_decoder.cpp [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+            ++point_cloud->width;
+        }
+    }
+    {
+        point_cloud_pub.publish(point_cloud);
+    }
+    return;
+}
+
+// void LslidarC16Decoder::publishPointCloud() {
```
Comment (fengqikai1414) [inline modules/drivers/lslidar_c16/lslidar_c16_decoder/src/lslidar_c16_decoder.cpp]: 删除无用的代码

**Comment by lianglia-apollo** [comment]
Please sign the CLA , thanks!

File: modules/drivers/lslidar_apollo/lslidar_compensator/include/lslidar_compensator/compensator.h [hunk 7]
```diff
@@ -0,0 +1,105 @@
+/***************************************************************************
+Copyright 2018 The Leishen Authors. All Rights Reserved                     /
```
Comment (lianglia-apollo) [inline modules/drivers/lslidar_apollo/lslidar_compensator/include/lslidar_compensator/compensator.h]: Please use "The Apollo Authors" in all the Copyright statements.

**Pull request merged**